server/selection.go

```diff
@@ -183,10 +183,10 @@ func (s *MinLSSelector) selectUnknownSession() *BroadcastSession {
 		totalStake += stake
 	}
 
-	// Generate a random stake weight between 0 and totalStake
 	r := int64(0)
+	// Generate a random stake weight between 1 and totalStake
 	if totalStake > 0 {
-		r = rand.Int63n(totalStake)
+		r = 1 + rand.Int63n(totalStake)
```

Comment:
When `totalStake > 0`, the min for `r` needs to be 1. Otherwise, if `r` is set to 0 and there are zero stake weight sessions before non-zero stake weight sessions in `unknownSessions`, then the zero stake weight sessions would be selected before the non-zero stake weight sessions.